src/utils/switch-site.ts

```diff
@@ -18,9 +18,13 @@ const SITE_TO_EMAIL_MAP: Record<SiteKey, string> = {
 export const setSiteCookie = (site: SiteKey) =>
   Cookies.set(COOKIE_USER_KEY, SITE_TO_EMAIL_MAP[site])
 
-/** Set the initial site if the demo is visited for the first time. */
-export function setInitialSiteCookie() {
-  if (!Cookies.get(COOKIE_USER_KEY)) setSiteCookie(DEFAULT_SITE)
+/** Sync the user cookie for the API with the site in localStorage. */
+export function syncInitialSiteCookie() {
+  // Use the site from localStorage as the initial user cookie.
+  // This is a migration for people who have previously visited the demo.
+  const site = JSON.parse(localStorage.getItem(SITE_KEY)!) || DEFAULT_SITE
```

Comment:
I think the problem might be because we're storing sites in both the cookies via `Cookies` and localStorage via `atomWithStorage`.

I don't quite understand why we have to have sites stored in 2 places. Would storing the site in only the atom work?

I found you [can modify `atomWithStorage` to store in cookies](https://github.com/pmndrs/jotai/issues/1152) instead, so we only have a single source of truth. What do you think?

Reply:
@WiNloSt Yep, using cookie alone should totally work. As noted in the PR description, this is mostly a hotfix I applied to the release branch to prevent the deployed Vercel demo from breaking:

> Yes, we could've also used the cookie as the single source of truth and get rid of the localStorage altogether, as they're redundant given the site maps neatly to a single user - to avoid them being out-of-sync. I might do that in a later PR.

I'll create a follow-up PR that gets rid of the `localStorage`. We can either use the `SiteKey` as a source of truth for the API as well (so we don't need to do any mapping), or map the SiteKey to the emails.